### ISSUE TITLE
Route::error not working

### ISSUE THREAD

**Issue by kztneda** [skipperbent/simple-php-router#551]
Hello.

IDK what I'm doing wrong but I can't get the error handling working well. I've placed all the routes as simple as possible to try the error get working but I can't. The normal routes work well just the error handling don't.

Here is my Routes file.
```PHP
  namespace Routes;
  use \Pecee\SimpleRouter\SimpleRouter as Router;
  
  include_once 'Controls/Errors.php';

  Router::get('/', function() {
      return 'Hello world';
  });
  Router::get('/{id}', function($id) {
      return $id;
  });
  Router::get('/App/{id}', function($id) {
      return $id;
  });
  
  Router::get('/not-found', 'Errors@notFound');
  Router::error(function(Request $request, \Exception $exception) {
      if($exception instanceof NotFoundHttpException && $exception->getCode() === 404) {
          response()->redirect('/not-found');
      }
  });
```

here is what I have on "Controls/Error.php"
```PHP
namespace Controls;

class Errors
{
    public function notFound():string{
        return "Not found";
    }
}
```

I always get this error instead of the error handle response I want:
`Fatal error: Uncaught Pecee\SimpleRouter\Exceptions\NotFoundHttpException: Route not found: "/asd/asd/s/" in **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php:450 Stack trace: #0 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php(339): Pecee\SimpleRouter\Router->routeRequest() #1 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\SimpleRouter.php(68): Pecee\SimpleRouter\Router->start() #2 **PATH TO LIB**\Pecee\SimpleRouter\SimpleRouter::start() #3 **MAIN PATH**\index.php(20): Router::read(false) #4 {main} thrown in **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php on line 450`

Here is my index.php
```PHP
require_once '**PATH TO LIB**/Router.php';
Router::read(false);
```

Here is '**PATH TO LIB**/Router.php'
```PHP
require_once '**PATH TO LIB**/vendor/autoload.php';
use \Pecee\SimpleRouter\SimpleRouter;

class Router extends SimpleRouter 
{
    public static function read($isRunningLocally)
    {
        $prefix = ($isRunningLocally) ? '/test' : null;
        static::group(['prefix' => $prefix], function() {
            require_once 'Routes.php';
        });
        parent::start();
    }
}
```

Could you please help me guys?.

**Comment by skipperbent** [comment]
Hi.

Thanks for the detailed issue. I understand the confusion as everything looks great. Did some digging and it looks like there's an issue with the `Router::error` helper when called from inside a group.

I will add an update that solves this issue momentarily and keep you notified once it's available.

/ Simon

**Comment by skipperbent** [comment]
Hi @kztneda.

I've provided a fix for the issue in version 4.3.4.1.

Thanks for taking the time to report the issue.
Feel free to reopen or create another issue if you have any questions, comments or problems.

/ Simon

**Comment by kztneda** [comment]
oh, now I get this error

`Fatal error: Uncaught TypeError: Router::Generales\{closure}(): Argument #1 ($request) must be of type Generales\Request, Pecee\Http\Request given in **MAIN PATH**\assets\php\Routes.php:26 Stack trace: #0 [internal function]: Router::Generales\{closure}(Object(Pecee\Http\Request), Object(Pecee\SimpleRouter\Exceptions\NotFoundHttpException)) #1 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Handlers\CallbackExceptionHandler.php(34): call_user_func(Object(Closure), Object(Pecee\Http\Request), Object(Pecee\SimpleRouter\Exceptions\NotFoundHttpException)) #2 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php(532): Pecee\SimpleRouter\Handlers\CallbackExceptionHandler->handleError(Object(Pecee\Http\Request), Object(Pecee\SimpleRouter\Exceptions\NotFoundHttpException)) #3 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php(450): Pecee\SimpleRouter\Router->handleException(Object(Pecee\SimpleRouter\Exceptions\NotFoundHttpException)) #4 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php(339): Pecee\SimpleRouter\Router->routeRequest() #5 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\SimpleRouter.php(68): Pecee\SimpleRouter\Router->start() #6 **MAIN PATH**\assets\php\Router.php(13): Pecee\SimpleRouter\SimpleRouter::start() #7 **MAIN PATH**\index.php(4): Router::read(true) #8 {main} thrown in **MAIN PATH**\assets\php\Routes.php on line 26`

the line 26 on my Routes.php is this
`Router::error(function(Request $request, \Exception $exception) {`

**Comment by skipperbent** [comment]
Make sure to import the Request class from simple-router or use the full class name otherwise php will get confused.

```php
Router::error(static function(\Pecee\Http\Request $request, \Exception $exception) {
```

**Comment by kztneda** [comment]
Hello @SimonSchobel, I actually did but it still shows the first error message.

`Fatal error: Uncaught Pecee\SimpleRouter\Exceptions\NotFoundHttpException: Route not found: "/asd/asd/s/" in **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php:450 Stack trace: #0 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php(339): Pecee\SimpleRouter\Router->routeRequest() #1 **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\SimpleRouter.php(68): Pecee\SimpleRouter\Router->start() #2 **PATH TO LIB**\Pecee\SimpleRouter\SimpleRouter::start() #3 **MAIN PATH**\index.php(20): Router::read(false) #4 {main} thrown in **PATH TO LIB**\vendor\pecee\simple-router\src\Pecee\SimpleRouter\Router.php on line 450
`

I have debugged the process and found that the ExceptionHandler is added in your `exceptionHandlers` Array in your `Router.php` file when I do the Router::error, but when the error occurs there are no ExceptionHandler on the queue, it disappeared! lol.

**Comment by skipperbent** [comment]
That's strange. I think the problem is that the exception is added within a group still.

Try one of these things and see if they help, it will help me pinpoint the error:

- Try different prefixes for the group where you call ::error and see if that renders the exception. Try prefix = '', prefix ='/' instead of prefix = null.

- Try moving error:: to the beginning of the group and see if that renders the exception.

Obviously the steps above is not ideal, but will help me locate if it's a problem with the prefix not rendering the group where your error:: call is located.

Thanks.

/ Simon

**Comment by skipperbent** [comment]
The unit-test below works.

```php
public function testExceptionHandlerCallback()
{
    TestRouter::group(['prefix' => null], function() {

        TestRouter::get('/', function() {
            return 'Hello world';
        });
        TestRouter::get('/{id}', function($id) {
            return $id;
        });
        TestRouter::get('/App/{id}', function($id) {
            return $id;
        });

        TestRouter::get('/not-found', 'Errors@notFound');

        TestRouter::error(static function(\Pecee\Http\Request $request, \Exception $exception) {

            if($exception instanceof \Pecee\SimpleRouter\Exceptions\NotFoundHttpException && $exception->getCode() === 404) {
                return $request->setRewriteCallback(static function() {
                    return 'success';
                });
            }
        });
    });

    $result = TestRouter::debugOutput('/non/existing', 'get');
    $this->assertEquals('success', $result);
}
```

What does your `SimpleRouter::error` (SimpleRouter line 426-435) function look like?

Can you try to replace `parent::start()` with the code below and paste the output here?

```php
$debugInfo = parent::startDebug();
echo sprintf('<pre>%s</pre>', var_export($debugInfo));
```

**Comment by kztneda** [comment]
Hi I've tried the path change, and the relocation for the Route but didn't worked.

About the `SimpleRouter::error` is this
```php
public static function error(Closure $callback): CallbackExceptionHandler
    {
        $callbackHandler = new CallbackExceptionHandler($callback);

        static::router()->addRoute(
            (new RouteGroup())->addExceptionHandler($callbackHandler)
        );

        return $callbackHandler;
    }
```

the output of the debug is this:
```json
array ( 'url' => Pecee\Http\Url::__set_state(array( 'originalUrl' => '/BSConnectWEB/asd', 'scheme' => NULL, 'host' => 'localhost', 'port' => NULL, 'username' => NULL, 'password' => NULL, 'path' => '/BSConnectWEB/asd/', 'params' => array ( ), 'fragment' => NULL, )), 'method' => 'get', 'host' => 'localhost', 'loaded_routes' => array ( ), 'all_routes' => array ( 0 => Pecee\SimpleRouter\Route\RouteGroup::__set_state(array( 'urlRegex' => '/^%s\\/?/u', 'prefix' => '/BSConnectWEB', 'name' => NULL, 'domains' => array ( ), 'exceptionHandlers' => array ( ), 'filterEmptyParams' => true, 'defaultParameterRegex' => NULL, 'paramModifiers' => '{}', 'paramOptionalSymbol' => '?', 'group' => NULL, 'parent' => NULL, 'callback' => Closure::__set_state(array( )), 'defaultNamespace' => NULL, 'namespace' => NULL, 'requestMethods' => array ( ), 'where' => array ( ), 'parameters' => array ( ), 'originalParameters' => array ( ), 'middlewares' => array ( ), )), ), 'boot_managers' => array ( ), 'csrf_verifier' => NULL, 'log' => array ( 0 => array ( 'message' => 'Router starting', 'time' => '0.0050470829', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\SimpleRouter.php', 'line' => 82, 'function' => 'start', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 1 => array ( 'message' => 'Loading routes', 'time' => '0.0087420940', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 327, 'function' => 'loadRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 2 => array ( 'message' => 'Processing routes', 'time' => '0.0087609291', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 285, 'function' => 'processRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 3 => array ( 'message' => 'Processing route "Pecee\\SimpleRouter\\Route\\RouteGroup"', 'time' => '0.0087878704', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 285, 'function' => 'processRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 4 => array ( 'message' => 'Starting rendering route "Pecee\\SimpleRouter\\Route\\RouteGroup"', 'time' => '0.0088520050', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 189, 'function' => 'renderRoute', 'class' => 'Pecee\\SimpleRouter\\Route\\Route', 'type' => '->', ), ), 5 => array ( 'message' => 'Parsing parameters', 'time' => '0.0088620186', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 189, 'function' => 'renderRoute', 'class' => 'Pecee\\SimpleRouter\\Route\\Route', 'type' => '->', ), ), 6 => array ( 'message' => 'Finished parsing parameters', 'time' => '0.0088729858', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 189, 'function' => 'renderRoute', 'class' => 'Pecee\\SimpleRouter\\Route\\Route', 'type' => '->', ), ), 7 => array ( 'message' => 'Executing callback', 'time' => '0.0088858604', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 189, 'function' => 'renderRoute', 'class' => 'Pecee\\SimpleRouter\\Route\\Route', 'type' => '->', ), ), 8 => array ( 'message' => 'Processing routes', 'time' => '0.0155370235', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 199, 'function' => 'processRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 9 => array ( 'message' => 'Processing route "Pecee\\SimpleRouter\\Route\\RouteUrl"', 'time' => '0.0155589581', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 199, 'function' => 'processRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 10 => array ( 'message' => 'Processing route "Pecee\\SimpleRouter\\Route\\RouteGroup"', 'time' => '0.0155909061', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 199, 'function' => 'processRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 11 => array ( 'message' => 'Starting rendering route "Pecee\\SimpleRouter\\Route\\RouteGroup"', 'time' => '0.0156519413', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 189, 'function' => 'renderRoute', 'class' => 'Pecee\\SimpleRouter\\Route\\Route', 'type' => '->', ), ), 12 => array ( 'message' => 'Processing route "Pecee\\SimpleRouter\\Route\\RouteUrl"', 'time' => '0.0156600475', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 199, 'function' => 'processRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 13 => array ( 'message' => 'Processing route "Pecee\\SimpleRouter\\Route\\RouteUrl"', 'time' => '0.0156769753', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 199, 'function' => 'processRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 14 => array ( 'message' => 'Finished loading routes', 'time' => '0.0156989098', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 327, 'function' => 'loadRoutes', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 15 => array ( 'message' => 'Routing request', 'time' => '0.0157060623', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 339, 'function' => 'routeRequest', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 16 => array ( 'message' => 'Matching route "Pecee\\SimpleRouter\\Route\\RouteUrl"', 'time' => '0.0157160759', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 339, 'function' => 'routeRequest', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 17 => array ( 'message' => 'Matching route "Pecee\\SimpleRouter\\Route\\RouteUrl"', 'time' => '0.0157470703', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 339, 'function' => 'routeRequest', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 18 => array ( 'message' => 'Matching route "Pecee\\SimpleRouter\\Route\\RouteUrl"', 'time' => '0.0157780647', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 339, 'function' => 'routeRequest', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 19 => array ( 'message' => 'Route not found: "/BSConnectWEB/asd/"', 'time' => '0.0158109665', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 339, 'function' => 'routeRequest', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 20 => array ( 'message' => 'Starting exception handling for "Pecee\\SimpleRouter\\Exceptions\\NotFoundHttpException"', 'time' => '0.0168340206', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 450, 'function' => 'handleException', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 21 => array ( 'message' => 'Processing exception-handler "Pecee\\SimpleRouter\\Handlers\\CallbackExceptionHandler"', 'time' => '0.0168569088', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 450, 'function' => 'handleException', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 22 => array ( 'message' => 'Start rendering exception handler', 'time' => '0.0168619156', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 450, 'function' => 'handleException', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 23 => array ( 'message' => 'Finished rendering exception-handler', 'time' => '0.0168790817', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 450, 'function' => 'handleException', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 24 => array ( 'message' => 'Finished processing', 'time' => '0.0168869495', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 450, 'function' => 'handleException', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), 25 => array ( 'message' => 'Finished exception handling - exception not handled, throwing', 'time' => '0.0168919563', 'trace' => array ( 'file' => 'D:\\xampp\\htdocs\\BSConnectWEB\\assets\\php\\vendor\\pecee\\simple-router\\src\\Pecee\\SimpleRouter\\Router.php', 'line' => 450, 'function' => 'handleException', 'class' => 'Pecee\\SimpleRouter\\Router', 'type' => '->', ), ), ), 'event_handlers' => array ( ), 'router_output' => NULL, 'library_version' => false, 'php_version' => '8.0.0', 'server_params' => array ( 'redirect_mibdirs' => 'D:/xampp/php/extras/mibs', 'redirect-mibdirs' => 'D:/xampp/php/extras/mibs', 'redirect_mysql_home' => '\\xampp\\mysql\\bin', 'redirect-mysql-home' => '\\xampp\\mysql\\bin', 'redirect_openssl_conf' => 'D:/xampp/apache/bin/openssl.cnf', 'redirect-openssl-conf' => 'D:/xampp/apache/bin/openssl.cnf', 'redirect_php_pear_sysconf_dir' => '\\xampp\\php', 'redirect-php-pear-sysconf-dir' => '\\xampp\\php', 'redirect_phprc' => '\\xampp\\php', 'redirect-phprc' => '\\xampp\\php', 'redirect_tmp' => '\\xampp\\tmp', 'redirect-tmp' => '\\xampp\\tmp', 'redirect_status' => '200', 'redirect-status' => '200', 'mibdirs' => 'D:/xampp/php/extras/mibs', 'mysql_home' => '\\xampp\\mysql\\bin', 'mysql-home' => '\\xampp\\mysql\\bin', 'openssl_conf' => 'D:/xampp/apache/bin/openssl.cnf', 'openssl-conf' => 'D:/xampp/apache/bin/openssl.cnf', 'php_pear_sysconf_dir' => '\\xampp\\php', 'php-pear-sysconf-dir' => '\\xampp\\php', 'phprc' => '\\xampp\\php', 'tmp' => '\\xampp\\tmp', 'http_host' => 'localhost', 'http-host' => 'localhost', 'http_connection' => 'keep-alive', 'http-connection' => 'keep-alive', 'http_pragma' => 'no-cache', 'http-pragma' => 'no-cache', 'http_cache_control' => 'no-cache', 'http-cache-control' => 'no-cache', 'http_sec_ch_ua' => '" Not A;Brand";v="99", "Chromium";v="90", "Microsoft Edge";v="90"', 'http-sec-ch-ua' => '" Not A;Brand";v="99", "Chromium";v="90", "Microsoft Edge";v="90"', 'http_sec_ch_ua_mobile' => '?0', 'http-sec-ch-ua-mobile' => '?0', 'http_upgrade_insecure_requests' => '1', 'http-upgrade-insecure-requests' => '1', 'http_user_agent' => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36 Edg/90.0.818.62', 'http-user-agent' => 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36 Edg/90.0.818.62', 'http_accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9', 'http-accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9', 'http_sec_fetch_site' => 'none', 'http-sec-fetch-site' => 'none', 'http_sec_fetch_mode' => 'navigate', 'http-sec-fetch-mode' => 'navigate', 'http_sec_fetch_user' => '?1', 'http-sec-fetch-user' => '?1', 'http_sec_fetch_dest' => 'document', 'http-sec-fetch-dest' => 'document', 'http_accept_encoding' => 'gzip, deflate, br', 'http-accept-encoding' => 'gzip, deflate, br', 'http_accept_language' => 'es-419,es;q=0.9,es-ES;q=0.8,en;q=0.7,en-GB;q=0.6,en-US;q=0.5', 'http-accept-language' => 'es-419,es;q=0.9,es-ES;q=0.8,en;q=0.7,en-GB;q=0.6,en-US;q=0.5', 'http_cookie' => 'PHPSESSID=1tmbb56h16nkf43haj66hkjgto', 'http-cookie' => 'PHPSESSID=1tmbb56h16nkf43haj66hkjgto', 'path' => 'C:\\Program Files\\Common Files\\Oracle\\Java\\javapath;C:\\Python39\\Scripts\\;C:\\Python39\\;C:\\Program Files (x86)\\Common Files\\Intel\\Shared Libraries\\redist\\intel64\\compiler;C:\\WINDOWS\\system32;C:\\WINDOWS;C:\\WINDOWS\\System32\\Wbem;C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\;C:\\WINDOWS\\System32\\OpenSSH\\;C:\\Program Files\\PuTTY\\;C:\\Program Files\\dotnet\\;C:\\Program Files\\Microsoft SQL Server\\130\\Tools\\Binn\\;C:\\Program Files\\Microsoft SQL Server\\Client SDK\\ODBC\\170\\Tools\\Binn\\;C:\\Program Files\\Git\\cmd;C:\\Program Files\\Common Files\\Autodesk Shared\\;C:\\Program Files\\Microsoft SQL Server\\120\\Tools\\Binn\\;C:\\Program Files\\nodejs\\;C:\\ProgramData\\chocolatey\\bin;C:\\ProgramData\\ComposerSetup\\bin;C:\\Program Files\\MySQL\\MySQL Shell 8.0\\bin\\;C:\\Users\\kovu1\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\kovu1\\AppData\\Local\\Programs\\Microsoft VS Code\\bin;C:\\Users\\kovu1\\.dotnet\\tools;C:\\php;C:\\Users\\kovu1\\AppData\\Local\\ComposerSetup\\bin;C:\\Users\\kovu1\\AppData\\Roaming\\Composer\\vendor\\bin;C:\\Users\\kovu1\\AppData\\Roaming\\npm;', 'systemroot' => 'C:\\WINDOWS', 'comspec' => 'C:\\WINDOWS\\system32\\cmd.exe', 'pathext' => '.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW', 'windir' => 'C:\\WINDOWS', 'server_signature' => '
Apache/2.4.46 (Win64) OpenSSL/1.1.1h PHP/8.0.0 Server at localhost Port 80
', 'server-signature' => '
Apache/2.4.46 (Win64) OpenSSL/1.1.1h PHP/8.0.0 Server at localhost Port 80
', 'server_software' => 'Apache/2.4.46 (Win64) OpenSSL/1.1.1h PHP/8.0.0', 'server-software' => 'Apache/2.4.46 (Win64) OpenSSL/1.1.1h PHP/8.0.0', 'server_name' => 'localhost', 'server-name' => 'localhost', 'server_addr' => '::1', 'server-addr' => '::1', 'server_port' => '80', 'server-port' => '80', 'remote_addr' => '::1', 'remote-addr' => '::1', 'document_root' => 'D:/xampp/htdocs', 'document-root' => 'D:/xampp/htdocs', 'request_scheme' => 'http', 'request-scheme' => 'http', 'context_prefix' => '', 'context-prefix' => '', 'context_document_root' => 'D:/xampp/htdocs', 'context-document-root' => 'D:/xampp/htdocs', 'server_admin' => 'postmaster@localhost', 'server-admin' => 'postmaster@localhost', 'script_filename' => 'D:/xampp/htdocs/BSConnectWEB/index.php', 'script-filename' => 'D:/xampp/htdocs/BSConnectWEB/index.php', 'remote_port' => '62204', 'remote-port' => '62204', 'redirect_url' => '/BSConnectWEB/asd', 'redirect-url' => '/BSConnectWEB/asd', 'gateway_interface' => 'CGI/1.1', 'gateway-interface' => 'CGI/1.1', 'server_protocol' => 'HTTP/1.1', 'server-protocol' => 'HTTP/1.1', 'request_method' => 'GET', 'request-method' => 'GET', 'query_string' => '', 'query-string' => '', 'request_uri' => '/BSConnectWEB/asd', 'request-uri' => '/BSConnectWEB/asd', 'script_name' => '/BSConnectWEB/index.php', 'script-name' => '/BSConnectWEB/index.php', 'path_info' => '/asd', 'path-info' => '/asd', 'path_translated' => 'D:\\xampp\\htdocs\\asd', 'path-translated' => 'D:\\xampp\\htdocs\\asd', 'php_self' => '/BSConnectWEB/index.php/asd', 'php-self' => '/BSConnectWEB/index.php/asd', 'request_time_float' => 1621450397.609246, 'request-time-float' => 1621450397.609246, 'request_time' => 1621450397, 'request-time' => 1621450397, ), )
```

**Comment by skipperbent** [comment]
Thanks for the debug info!

Looks like the group is rendered correctly. So that's all good.

Try moving the call to `::error` above `parent::render();` and see if it triggers the exceptionhandler.

**Comment by kztneda** [comment]
`parent::render()` ? I guess you mean `parent::start()`.

If so, I did but the same.

I think I'll handle the errors with the .htaccess file.

**Comment by skipperbent** [comment]
> `parent::render()` ? I guess you mean `parent::start()`.
> 
> If so, I did but the same.
> 
> I think I'll handle the errors with the .htaccess file.

Yes, I'm sorry for the confusion.

I'm creating a patch for the issue as we speak. 
I would applicate if you could test if it works if you move the line out of the group.

Thanks.

/ Simon

**Comment by skipperbent** [comment]
Please check if the issue has been resolved in version 4.3.4.2.

Thanks.

/ Simon
